src/components/Identity/createComponent.js

```diff
@@ -143,7 +143,9 @@ export default (processIdSyncs, config, logger, optIn, eventManager) => {
     },
     commands: {
       setCustomerIds(options) {
-        return optIn.whenOptedIn().then(() => customerIds.sync(options));
+        return optIn
+          .whenOptedIn()
+          .then(() => customerIds.sync(options, logger));
```

Comment:
Hey @jileeshs, I just noticed you're passing through `logger` when you call `sync`. Instead, you should pass it through [when `customerIds` gets created](https://github.com/adobe/alloy/pull/319/files#diff-e3608b4e8d26a7ee90affbadd009744eR29) since the logger won't change across `sync` calls.

Reply:
Logged here: https://jira.corp.adobe.com/browse/CORE-39144